infra/Pos/Diffusion/Subscription/Common.hs

```diff
@@ -60,6 +60,7 @@ subscribeTo
     => Timer -> SendActions m -> NodeId -> m SubscriptionTerminationReason
 subscribeTo keepAliveTimer sendActions peer = do
     logNotice $ msgSubscribingTo peer
+    -- 'try' is from safe-exceptions, so it won't catch asyncs.
```

Comment:
Yep. This is fine for now. Ultimately we should probably review our exception handling strategy here and e.g. distinguish between `IOException`s and "expected" protocol errors.